app/CommandExecutor.php

```diff
@@ -13,6 +13,21 @@ class CommandExecutor
     /** @var array $commands */
     protected $commands = [];
 
+    /** @var string $outputFileName */
+    protected $outputFileName = '';
+
+    public function __construct()
+    {
+        $this->outputFileName = @tempnam(sys_get_temp_dir(), 'fwd_output_');
+    }
+
+    public function __destruct()
+    {
+        if ($this->outputFileName) {
```

Comment:
Now that we moved  the file name definition to the `__construct`, we can trust it is defined... However, we must check then if the file indeed exists in order to call `unlink`, to maybe change this `if` to `file_exists()`.

Reply:
i think we can just do it... if it breaks on file creation, the destruct wont be called, right?